drv/cosmo-hf/src/apob.rs

```diff
@@ -0,0 +1,124 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Tools to extract the APOB location from an AMD ROM
+//!
+//! For details, see AMD document 57299; tables and sections in this code refer
+//! to Rev. 2.0 February 2025.
+
+use crate::hf::ServerImpl;
+use drv_hf_api::HfError;
+use userlib::UnwrapLite;
+use zerocopy::{FromBytes, Immutable, IntoBytes};
+
+/// Embedded firmware structure (Table 3)
+///
+/// Only relevant fields are included here; the EFS extends beyond the size of
+/// this `struct`, but we don't care about any subsequent fields.
+#[derive(FromBytes, Immutable, IntoBytes)]
+#[repr(C)]
+pub struct Efs {
```

Comment:
Any reason didn't just use the definitions from [amd-efs](https://github.com/oxidecomputer/amd-efs)? It should be no_std as I believe @daym intended to make it usable from hubris.